### DISCUSSION TITLE
TypeScript Source

### DISCUSSION THREAD

**Discussion by Paganiniana** [browserengineering/book#904] [Show and tell]
Hey folks! I'm seeing a few conversations happening here about source code. Just wanted to share. I'm working on a TS implementation that should allow someone to see the project built, progressively, in a browser. (Obviously, there's a little bit of handwaving around the sockets in Chapter 1, though the code is there to run it via Node). Here's [the repo,](https://github.com/Paganiniana/fp-browser) if it proves useful for anyone!

**Comment by pavpanchekha** [comment]
Looks like great progress! Happy to see people reading the book, especially in other languages. We consciously tried to avoid complex Python-isms when possible; if you run into anything that’s hard to translate, let us know!
